mappings/net/minecraft/entity/Entity.mapping

```diff
@@ -80,6 +81,7 @@ CLASS aif net/minecraft/entity/Entity
 	METHOD U getScoreboardTags ()Ljava/util/Set;
 	METHOD V kill ()V
 	METHOD W getDataTracker ()Lqk;
+	METHOD X putOnTopmostSolidBlock ()V
```

Comment:
But this method is overriden in subclasses and does more than just move the entity to the top.

Reply:
Enigma doesn't show anything. ![grafik](https://user-images.githubusercontent.com/12124394/56890398-9e3e9780-6a79-11e9-9db5-c3e8fcd90a45.png)
That is the method I mapped here:
```java
@Environment(EnvType.CLIENT)
protected void putOnTopmostSolidBlock() {
	if (this.world == null) {
		return;
	}
	while (this.y > 0.0 && this.y < 256.0) {
		this.setPosition(this.x, this.y, this.z);
		if (this.world.doesNotCollide(this)) {
			break;
		}
		++this.y;
	}
	this.setVelocity(Vec3d.ZERO);
	this.pitch = 0.0f;
}
```

Reply:
Hmm, that doesn't seem to be used for other entities than players at all. What is it doing in `Entity`?

Reply:
What really confuses me now is that it's called in `onGameJoin`. When you join, your position doesn't get changed, nor does it reset the health. Or is that method only executed when the player *first* joins a server?

Reply:
The method is only executed client-side every time a player joins or respawns. The health is probably sent in another packet after the respawn packet.

Reply:
`respawn` (or maybe `onRespawn`?) would be a better name for it.

Reply:
It's currently called afterRespawn. Is that okay?